### PR TITLE
Add summarize-long-text extension

### PR DESCRIPTION
**Pull request by ajk68**
## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by ajk68** [comment]
Not an experienced developer and first Typescript/React project. Looking forward to your feedback.

**Comment by pernielsentikaer** [comment]
Thanks for taking the time to contribute! 🔥

How does this compare to the other extensions that do the same?

**Comment by ajk68** [comment]
This summarizes text that is too long for the context window of gpt. It
summarizes recursively. It is like the summarizes youtube plugin but then
for text in general.

I’m not aware of a plugin that does the same, but lmk if i missed something

Bye, arthur

File: extensions/summarize-long-text/package.json [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      ],
+      "required": false
+    },
+    {
+      "name": "openaiApiToken",
+      "title": "OpenAI API Token",
+      "description": "Your OpenAI API Token. Required if you choose ChatGPT as your AI.",
+      "type": "password",
+      "default": "",
+      "required": false
```
Comment (pernielsentikaer) [inline extensions/summarize-long-text/package.json:79]:
```suggestion
      "required": true
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- This is no toast when there are wrong (or none) credentials

summarize-long-text 2023-10-16 at 13 16 37


I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by ajk68** [comment]
Hi Per,

Sorry for delay. Was abroad in very remote place. Thanks for your feedback.

I didn't add a toast for this case as i thought it was too much information
for the small and temporary toast, but there is this screen when there is
no openai api token. Didn't you see this?

I didn't want to make the opanai key mandatory as you suggested, because
the default is using Raycast AI, and i didn't want to force users who
already paid for raycast AI to also pay for an openai key.

Can you let me know what you think?

Bye, Arthur


[image: Screenshot 2023-10-31 at 13.06.28.png]

File: extensions/summarize-long-text/src/summarizeBlocks.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            title="Continue in Chat"
+            icon={Icon.SpeechBubble}
+            url={`raycast://extensions/raycast/raycast-ai/ai-chat?fallbackText=${encodeURIComponent(
+              summary as string
+            )}`}
+          />
+        </ActionPanel>
+      }
+      isLoading={isLoading}
+      markdown={summary}
```
Comment (pernielsentikaer) [inline extensions/summarize-long-text/src/summarizeBlocks.tsx:54]: In both commands, maybe you can show some other content in <Detail /> if an error occurs, it feels a bit empty 🙂

**Pull request closed without merging**